lib/gui/components/safe-webview/safe-webview.js

```diff
@@ -0,0 +1,215 @@
+     * @public
+     */
+    this.loadingTimeoutId = null;
+    this.state = {
+      shouldLoad: true
+    };
+
+    // Events steal 'this'
+    this.didStartLoading = _.bind(this.didStartLoading, this);
+    this.didFailLoad = _.bind(this.didFailLoad, this);
+    this.didStopLoading = _.bind(this.didStopLoading, this);
+    this.didGetResponseDetails = _.bind(this.didGetResponseDetails, this);
+  }
+
+  /**
+   * @returns {react.Element}
+   */
+  render() {
+    if (this.state.shouldLoad) {
+      const url = new URL(this.props.src);
+
+      // We set the 'etcher-version' GET parameter here.
+      url.searchParams.set('etcher-version', packageJSON.version);
+
+      return react.createElement('webview', {
+        ref: 'webview',
+        src: url.href
+      }, []);
+
+    }
+
+    // We have to return null explicitly, undefined is an error in React.
+    return null;
+  }
+
+  /**
+   * @summary Add the Webview events if there is an element
+   */
+  componentDidMount() {
+
+    // There is no element to add events to if 'shouldLoad' is false.
+    if (this.state.shouldLoad) {
+
+      // Events React is unaware of have to be handled manually
+      this.refs.webview.addEventListener('did-start-loading', this.didStartLoading);
+      this.refs.webview.addEventListener('did-fail-load', this.didFailLoad);
+      this.refs.webview.addEventListener('did-stop-loading', this.didStopLoading);
+      this.refs.webview.addEventListener('did-get-response-details', this.didGetResponseDetails);
+      this.refs.webview.addEventListener('new-window', this.constructor.newWindow);
+      this.refs.webview.addEventListener('console-message', this.constructor.consoleMessage);
+    }
+  }
+
+  /**
+   * @summary Remove the Webview events if there is an element
+   */
+  componentWillUnmount() {
+
+    // There is no element to remove events from if 'shouldLoad' is false.
+    if (this.state.shouldLoad) {
+
+      // Events React is unaware of have to be handled manually
+      this.refs.webview.removeEventListener('did-start-loading', this.didStartLoading);
+      this.refs.webview.removeEventListener('did-fail-load', this.didFailLoad);
+      this.refs.webview.removeEventListener('did-stop-loading', this.didStopLoading);
+      this.refs.webview.removeEventListener('did-get-response-details', this.didGetResponseDetails);
+      this.refs.webview.removeEventListener('new-window', this.constructor.newWindow);
+      this.refs.webview.removeEventListener('console-message', this.constructor.consoleMessage);
+
+      // Reset timeout id
+      this.loadingTimeoutId = null;
+    }
+  }
+
+  /**
+   * @summary Start the loading timeout timer
+   */
+  didStartLoading() {
+    if (this.props.timeout && this.loadingTimeoutId === null) {
+      this.loadingTimeoutId = setTimeout(() => {
+        this.setState({
+          shouldLoad: false
+        });
+
+      // Milliseconds to wait before cancelling the page-load.
+      }, this.props.timeout);
```

Comment:
Is that already handled by `if (this.props.timeout`? i.e. the code isn't triggered currently, but is still there if we want to re-enable it in future?